functions/fish_prompt.fish

```diff
@@ -895,9 +895,36 @@ function __bobthefish_prompt_desk -S -d 'Display current desk environment'
     set_color normal
 end
 
+function __bobthefish_prompt_find_file_up_re -S -d 'Find file(s) (as a regex), going up the parent directories'
+    set -l dir "$argv[1]"
+    set -l file "$argv[2]"
+
+    if [ -z "$dir" ]
+        return 1
+    end
+
+    while [ "$dir" ]
+        if string match -q -r "$dir/($file)\$" -- $dir/*
```

Comment:
You'd need to regex escape the `$dir` bit, since it's perfectly reasonable to have reserved regex characters in paths. 

You might be better off putting the matching in your $dir/* expansion instead? e.g. `$dir/{.nvmrc,.node-version,package.json}`.

Reply:
I just reworked the function to accept multiple values. I wanted to be clever with the glob/regex version but it's over engineering. PTAL.